### DISCUSSION TITLE
Can't copy data from key() into a variable?

### DISCUSSION THREAD

**Discussion by IssaAboudi** [nlohmann/json#3235] [Q&A] [answered]
newFile has a filepath to a json file. students a vector of student (student is a struct with fields of type strings - name, subjects etc. )

I can print out the key fine but I want to store it in the vector. It seems that x.key() is returning a string, and I can store it in a string but i cannot store it in the member variable of students - which is a string? I am so confused

```c++
nlohmann::json tempJSON = nlohmann::json::array();
newFile >> tempJSON;
for(int i = 0; i < tempJSON.size(); i++) {
    for(auto &x : tempJSON.at(i).items()){
        students[i].name = x.key();
        std::cout << x.key() << std::endl;
    }
}

**Comment by nlohmann** [comment, marked as answer]
What is the error message?

**Reply by IssaAboudi** [reply]
This is all I see in the terminal:
`Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)`

**Reply by IssaAboudi** [reply]
Ok actually I'm just dumb, there's nothing in the vector so my problem is that I'm accessing memory that doesn't exist T_T

I created a temporary student variable to store all the data into and it works perfectly fine. totally user error lol
